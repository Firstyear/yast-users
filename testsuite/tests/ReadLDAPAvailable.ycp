/**
 * File		: ReadLDAPAvailable.ycp
 * Module	: Users configurator
 * Summary	: Test of UsersLDAP::ReadAvailable
 * Authors	: Jiri Suchomel <jsuchome@suse.cz>
 */

{
    // testedfiles: UsersLDAP.pm

    map READ = $[
	"etc" : $[
	    "nsswitch_conf" : $[
		"passwd"	: "ldap files"
	    ],
	],
    ];

    import "Testsuite";
    import "UsersLDAP";

    Testsuite::Test (``(UsersLDAP::ReadAvailable ()), [READ, $[], $[]], 0);

    READ["etc","nsswitch_conf","passwd"]	= "files";

    Testsuite::Test (``(UsersLDAP::ReadAvailable ()), [READ, $[], $[]], 0);

    READ["etc","nsswitch_conf","passwd"]	= "sss";

    Testsuite::Test (``(UsersLDAP::ReadAvailable ()), [READ, $[], $[]], 0);

}
