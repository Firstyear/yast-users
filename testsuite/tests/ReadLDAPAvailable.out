Read	.etc.nsswitch_conf.passwd "ldap files"
Return	true
Read	.etc.nsswitch_conf.passwd "files"
Return	false
Read	.etc.nsswitch_conf.passwd "sss"
Return	true
